### ISSUE TITLE
getting 'could not reach the server' error

### ISSUE THREAD

**Issue by settur1409** [Chainlit/chainlit#845]
**Describe the bug**
I am trying to bringup a simple chatbot from chainlit. I am using local LLM. Hence there is some time delay in response. My problem is that, very frequently I am getting "could not reach the server" error. though I made LLM call to run via syncio, I am still facing same issue. Can some one help me on this. Below is the code that I am using to get response from llm

**To Reproduce**
running below code.
@cl.on_message
async def main(message: cl.Message):
    await cl.Message(
        content=f"Received: {message.content}",
    ).send()

    print("llm invoked, wait for response")
    query = message.content
    response = asyncio.run(agent_executor.invoke(input=query))
    print("response received", response)
    await cl.Message(
        content=f"Bot Answer: {response}"
    ).send()

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] windows
 - Version [e.g. 22] 1.0.401 version of chainlit

**Additional context**

**Comment by saadjelbini** [comment]
Did you managed to resolve the problem?

**Comment by settur1409** [comment]
Hey, I don't know if that's a good fix or bad, I added a while True in main file so, the chainlit instance didn't get killed. 
I continued with that, rather digging so much. I was expecting the service to run, help me if there is someother (better way) to keep the service running.

**Comment by khadar1020** [comment]
yes @settur1409 I am also facing the same problem while running the llama3 model

**Comment by abhinav-901** [comment]
I am facing the issue while using chromadb and openai in aws eks .. unable to find the fix..

**Comment by settur1409** [comment]
I suspect, its the issue with chainlit and nothing related to LLM, or vector store, as a work around you can have a while True in main file.

**Comment by httplups** [comment]
Can you share how do you put this while True? It was on the on_message function? thanks @settur1409

**Comment by settur1409** [comment]
Hope this should help

imports ...

@on_chat_start
async def chat_start()
     ...

@cl.on_message
async def main()
   wait for user input
   call chain invoke

if __name__=="__main__":
    while True:
        pass

**Comment by chrisputzu** [comment]
> Hope this should help
> 
> imports ...
> 
> @on_chat_start async def chat_start() ...
> 
> @cl.on_message async def main() wait for user input call chain invoke
> 
> if **name**=="**main**": while True: pass

I hope this helps others facing the same error. I resolved the issue by using concurrent.futures, which creates a separate thread to handle the task.

Below is a simple code example demonstrating how to run a text_to_speech function that takes time to execute. This solution resolved the 'Could not reach the server' error for me:

import asyncio
import concurrent.futures

async def speak_async(answer: str):
loop = asyncio.get_running_loop()
with concurrent.futures.ThreadPoolExecutor() as pool:
await loop.run_in_executor(pool, text_to_speech, answer)
